### ISSUE TITLE
Issues noted with 2.3

### ISSUE THREAD

**Issue by canonet28** [Jamee999/cricket#28]
Good job with the simulator overall. But as I was trying my hands with version 2.3 playing customleague, came across these:

1. The simulator is apparently simulating twice the number of entered matches. So if I enter 1 as the number of matches, it will simulate the match twice and results in two scorecards (scorecard1.txt and scorecard2.txt) inside the scorecards folder. Similarly, a custom league with 3 matches results in 6 different scorecards.

2. As expected with these twice number of matches being played, the stats csv files are messed up. However, the stats files seem to have their own separate issues. Some of the obvious ones I noticed are listed below:
        i. For both bowlstats.csv and batstats.csv files, the formatting of the table goes wonky at times with wrong data appearing in the wrong column. I think it mainly stems from the fact that the "Result" column varying takes up more than 1 cell, sometimes 2 cells, some other times 3 cells.    
       ii. In bowlstats.csv, the "Balls" column populates with 0 for most bowlers. Also if a bowler has incompleted overs at the end of the innings, the incompleted part of the over appears in the "Balls" column rather than "Overs" column. Eg, someone bowls 18.4 overs, 18 shows up under Overs column and 4 under Balls column. I dont know if it was meant that way. 
        iii. allstats.csv file: I would assume the aggregate stats come from the matches that we simulate. However, a few of the stats seem out of place: the number of innings, total runs, total balls faced, total balls bowled, maidens, runs conceded and wickets taken.

I hope those are easy fixes.

**Comment by Jamee999** [comment]
1 -  The input for leagues is somewhat wonky at the moment. When you input the number of games you want the teams to play, it runs a home **and away** round robin that number of times. I will look into changing it, ideally so that teams actually only play each other the number of times they should, less ideally so that the input on the screen matches what is happening.

2i -  D'oh. The issue in the statsfile is a dumb mistake by me. Because there are commas in the result description, the data is getting split by them. I can fix that, probably by splitting the results field into which team won, and the margin.

2ii - This is working as intended, although I don't know how much I like it.

2 iii - I'll look into this. 

Thanks so much for your feedback and bug spotting, I really appreciate it.

**Comment by canonet28** [comment]
Ahh...that's home and away. In a way that makes sense for a league. Thanks for the clarification!

Anyway, very excited about the overall development of cricketpy. Thanks for your work, only very happy to play along and help out with any feedback. Quite a few exciting possibilities; will be following actively in reddit and here.

**Comment by Jamee999** [comment]
Ok, I think I have fixed the problem with the stats. For some reason, in the first innings a player was involved in (possibly just in leagues?) they were having the whole team's batting innings assigned to their stats, rather than just their own batting. (I'm still not entirely happy with the nuts-and-bolts of _why_ it was happening, but I think I have changed it and it doesn't happen any more.)

I'm running the _altcricket_ file through to check that things look okay, which takes a while on my little laptop, but if everything looks okay, I should have a fix up soon.
